src/cdk/table/table.ts

```diff
@@ -174,6 +162,31 @@ export class CdkTable<T> implements CollectionViewer {
     }
   }
 
+  ngOnInit() {
+    // TODO(andrewseguin): Setup a listener for scrolling, emit the calculated view to viewChange
+    this._dataDiffer = this._differs.find([]).create(this._trackByFn);
+  }
+
+  ngAfterContentInit() {
+    // Initialize the column definitions map for easy lookup. Update the map when changed.
+    this._cacheColumnDefinitionsByName();
+    this._columnDefinitions.changes.subscribe(() => this._cacheColumnDefinitionsByName());
+  }
+
+  ngAfterContentChecked() {
+    this._renderUpdatedColumns();
+  }
+
+  ngAfterViewInit() {
+    this._isViewInitialized = true;
+  }
+
+  ngDoCheck() {
+    if (this._isViewInitialized && this.dataSource && !this._renderChangeSubscription) {
```

Comment:
Can you add a comment that explains what you're doing with `DoCheck` and why it's necessary?